### PR TITLE
Unit Test IssueCommentTable dimensions

### PR DESCRIPTION
**Pull request by BrianLitwin**
I'm trying to sneak up on  #1790 (supporting images in markdown tables).. wrote these tests as a starting point.

### CODE REVIEW

**Comment by BasThomas** [comment]
I love your testing mindset @BrianLitwin, thanks do much!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: FreetimeTests/IssueCommentTableTests.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+import XCTest
+import StyledText
+@testable import Freetime
+
+class IssueCommentTableTests: XCTestCase {
+    
+    func test_tableColumnDimensions() {
+        let contentSizeCategory = UIApplication.shared.preferredContentSizeCategory
```
Comment (BasThomas) [inline FreetimeTests/IssueCommentTableTests.swift]: Just for readability, can we pull out all this setup logic into the `setUp` or similar? Makes the test a bit easier to read and scan.

File: FreetimeTests/IssueCommentTableTests.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        let rowOne =    [ textItem,   textItem,       textItem       ]
+        let rowTwo =    [ textItem,   medTextItem,    textItem       ]
+        let rowThree =  [ textItem,   textItem,       longerTextItem ]
+        let rowFour =   [ tallerItem, textItem,       textItem       ]
+        
+        let rows = [rowOne, rowTwo, rowThree, rowFour]
+        var buckets = [TableBucket()]
+        var rowHeights = [CGFloat]()
+        
+        rows.forEach({ fillBuckets(rows: $0,
```
Comment (BasThomas) [inline FreetimeTests/IssueCommentTableTests.swift]: Nit: for readability, you can omit the `(` and `)` at the start and end.

**Comment by BrianLitwin** [comment]
Intent is to get up a list of tests that document the unique table considerations, such as: 

- cell sizes depend on the tallest cell in a row and widest cell in a column, determined in [this file](https://github.com/GitHawkApp/GitHawk/blob/master/Classes/Issues/Comments/Markdown/IssueCommentTableModel%2BModels.swift), the implication being that after an image loads, updating the table would require re-calculating those values  
- filling alternate table rows with a background color, calculated [here](https://github.com/GitHawkApp/GitHawk/blob/c287075ba6b8d34b07aed5890c165c7325f6cb13/Classes/Issues/Comments/Markdown/CMarkParsing.swift#L231-L238) and set [here](https://github.com/GitHawkApp/GitHawk/blob/c287075ba6b8d34b07aed5890c165c7325f6cb13/Classes/Issues/Comments/Markdown/CMarkParsing.swift#L212). For my purposes undermining the thought that you could just convert cell text to any old StyledTextRenderer. 
- the header row has special text traits set [here](https://github.com/GitHawkApp/GitHawk/blob/c287075ba6b8d34b07aed5890c165c7325f6cb13/Classes/Issues/Comments/Markdown/CMarkParsing.swift#L204)
- cell borders are set [here](https://github.com/GitHawkApp/GitHawk/blob/c287075ba6b8d34b07aed5890c165c7325f6cb13/Classes/Issues/Comments/Tables/IssueCommentTableCell.swift#L91-L92)

Both for reference and regression-proofing.

File: FreetimeTests/IssueCommentTableTests.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        let contentSizeCategory = UIApplication.shared.preferredContentSizeCategory
+        
+        let textItem =       StyledTextRenderer(string: StyledTextString(styledTexts: [StyledText(text: "item")]),
+                                                contentSizeCategory: contentSizeCategory)
+        let medTextItem =    StyledTextRenderer(string: StyledTextString(styledTexts: [StyledText(text: "medium item")]),
+                                                contentSizeCategory: contentSizeCategory)
+        let longerTextItem = StyledTextRenderer(string: StyledTextString(styledTexts: [StyledText(text: "longer text length item")]),
+                                                contentSizeCategory:contentSizeCategory)
+        let tallerItem =     StyledTextRenderer(string: StyledTextString(styledTexts: [StyledText(text: "T")]),
+                                                contentSizeCategory: .accessibilityExtraExtraLarge)
```
Comment (rnystrom) [inline FreetimeTests/IssueCommentTableTests.swift]: style nit: can we format this all w/ params in newlines? e.g.

```swift
let foo = Object(
  param: "bar",
  param2: "baz"
)
```

**Comment by BrianLitwin** [comment]
@BasThomas @rnystrom Thank you for your comments! The most recent commit is tidied up I believe.. let me know if there is more I can do.

**Pull request merged**